feature/amazon/src/test/java/com/revenuecat/purchases/amazon/StoreProductTest.kt

```diff
@@ -0,0 +1,54 @@
+package com.revenuecat.purchases.amazon
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.revenuecat.purchases.ProductType
+import com.revenuecat.purchases.amazon.helpers.dummyAmazonProduct
+import com.revenuecat.purchases.amazon.helpers.stubStoreProductForAmazon
+import com.revenuecat.purchases.models.GoogleStoreProduct
+import com.revenuecat.purchases.models.GoogleSubscriptionOption
+import com.revenuecat.purchases.models.Period
+import com.revenuecat.purchases.models.Price
+import com.revenuecat.purchases.models.PricingPhase
+import com.revenuecat.purchases.models.PurchasingData
+import com.revenuecat.purchases.models.RecurrenceMode
+import com.revenuecat.purchases.models.SubscriptionOptions
+import com.revenuecat.purchases.utils.mockProductDetails
+import org.assertj.core.api.AssertionsForInterfaceTypes.assertThat
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class StoreProductTest {
+
+    @Test
+    fun `Two StoreProducts with the same properties are equal and have same hashcode`() {
+        val product1 = dummyAmazonProduct().toStoreProduct("US")!!.amazonProduct
+        val product2 = dummyAmazonProduct().toStoreProduct("US")!!.amazonProduct
+        assertThat(product1).isEqualTo(product2)
+        assertThat(product1.hashCode()).isEqualTo(product2.hashCode())
```

Comment:
this is failing and i'm a little confused why, so i pushed for feedback. shouldn't the AmazonStoreProduct data class auto-generate a field-by-field equals, looking at all the fields in the primary ctor? i'm not seeing why any of those fields would fail equality check...

Reply:
This is the same why you had to add https://github.com/RevenueCat/purchases-android/pull/866/files#diff-595d4f18a593d2fa71649b5dda88680f1924f3628ca3b2b19e3e3b1e437112e7R119 . As @tonidero is mentioning, the JSONObject is not being compared correctly and should be excluded of the comparison. I don't think converting it to a string would work since I think the order of the JSON keys is not deterministic.

Reply:
ahhh my bad i should've remembered this. when inspecting the different fields, the jsonobject looked the same... thanks both!